ansible/config_sonic_basedon_testbed.yml

```diff
@@ -0,0 +1,88 @@
+# This Playbook run time generate matching configuration file for SONiC switch(Minigraph) based on a specific testbed topology specified in testbed.csv  
+# When user call testbed-cli to deploy a testbed topology, use this playbook to generate matching SONiC minigraph file and deploy it into SONiC switch under test. 
+# Or when you know your topology name, you may use this playbook alone to generate a minigraph matching your topology name without deploy it.
+#
+# VM Topologies are defined inside of vars/ directory in files vars/topo_{{ topology_name}}.yml
+#  Every topology should have a name to distinct one topology from another on the server
+#  Every topology contains a ptf container which will be used as placeholder for the injected interfaces from VMs, or direct connections to PTF host
+# VMs inventory file is also required to have all VMs ready for generating the minigraph file
+# VMs inventory is in file 'veos'
+#
+# Template files for generating minigraph.xml are defined in template/topo directory
+#
+# To generate and deploy minigraph for SONiC switch matching the VM topology please use following command
+# ANSIBLE_SCP_IF_SSH=y ansible-playbook -i lab config_sonic_basedon_testbed.yml --vault-password-file=~/.password -l sonic_dut_name -e VM_base=VM0300 -e topo=t0 [-e deploy=true]
+#
+# Parameters
+# -l str-msn2700-01          - the sonic_dut_name you are going to generate minigraph for
+# -e VM_base=VM0300          - the VM name which is used to as base to calculate VM name for this set
+# -e topo=t0                 - the name of removed topo
+# -e deploy=True             - if deploy the newly generated minigraph to the targent DUT, default is false if not defined
+#
+# After minigraph.xml is generated, the playbook will replace the original minigraph file under ansible/minigraph/ with the newly generated minigraph file for the SONiC device.
+# The playbook will based on deploy=True or False to deside if load the SONiC device with new minigraph or not. 
+# If deploy=True, the playbook will reboot the SONiC switch after change to new minigraph to make it take effect
+#
+####################################################################################################################################################################################
+
+- hosts: sonic
+  gather_facts: yes
+  tasks:
+  - fail: msg="need to provide topology type and vm base like topo=t0 and vm_base=VM100"
+    when: topo and vm_base is not defined 
```

Comment:
I think you need to check topo and vm_base in separate
I'd write it as
(topo is not defined) and (vm_base is not defined)